lib/random.gi

```diff
@@ -201,18 +203,18 @@ InstallMethod(Reset, [IsMersenneTwister, IsObject], function(rs, seed)
   return old;
 end);
 
-InstallMethod(Random, [IsMersenneTwister, IsList], function(rs, list)
-  return list[Random(rs, 1, Length(list))];
-end);
```

Comment:
@ChrisJefferson I have a vague memory that when you refactored this code last, we had to keep some methods (for ... something...?) instead of generic methods due to some issues that I forgot about... I am hoping this is not relevant here (in particular, I believe the tests for Random that you wrote would hopefully catch any such issues), but perhaps you could have a quick look at the changes in this PR just in case you remember this better than I do... :-)

Reply:
At some point when I worked on this I tried to make sure that GAP would produce the same sequence of random numbers on different versions -- now I think that wasn't worth the effort, as for most types we don't make that promise anyway. In some cases I found infinite loops in packages if I removed the wrong function, but if this passes all package tests then I think it's OK.

Reply:
I see the situation as follows.

We have generic `Random` methods for the cases `[ IsRandomSource, IsList and ... ]` and `[ IsRandomSource, IsInt, IsInt ]`.
They delegate to each other, with the effect that for specific kinds of random sources, one has to install only one such method. (If one does not install any `Random` method for a new kind of random sources, one runs into infinite loops, thus the generic methods are to some extent dangerous. However, we do not support "generic random sources".)
The `Random` methods for different specific kinds of random sources do not compete with each other, thus only the specific method (if available) and the generic method are applicable.

In the case of `IsMersenneTwister`, the specific method for `IsList` which I propose to remove does the same as the generic method for this case, thus the removal should not change anything at runtime, in particular the sequence of random numbers should not be affected.